server/core/storage.go

```diff
@@ -501,10 +503,46 @@ func (s *Storage) RemoveServiceGCSafePoint(serviceID string) error {
 	return s.Remove(key)
 }
 
-func (s *Storage) initServiceGCSafePointForGCWorker() (*ServiceSafePoint, error) {
+// fixGCWorkerServiceSafepoint tries to fix gc_worker's special service safepoint if it's missing or incorrect. An issue
+// in the older version may have invalid TTL for gc_worker's safepoint, and it also might be missing. gc_worker's
+// safepoint may also be missing when the cluster is just bootstrapped. Detect these cases and fix gc_worker's safepoint
+// if necessary.
+func (s *Storage) fixGCWorkerServiceSafePpoint(allServiceSafePoints []*ServiceSafePoint) (modified bool, err error) {
+	if len(allServiceSafePoints) == 0 {
+		// It's a new cluster, or everything is lost so we have no way to recover it. Initialize gc_worker's service
+		// safepoint to zero.
+		_, err = s.initServiceGCSafePointForGCWorker(0)
+		return true, err
```

Comment:
What if the initialization is wrong, we still return true?

Reply:
Yes but I thought err will be not nil here so it doesn't matter 🤔 maybe  it doesn't look good. I'll change it.